src/worksheet.cc

```diff
@@ -223,3 +229,157 @@ Napi::Value Worksheet::WriteURL(const Napi::CallbackInfo& info) {
                       Format::Get(info[3]));
   return env.Undefined();
 }
+
+Napi::Value Worksheet::Autofilter(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+  worksheet_autofilter(worksheet,
+                       info[0].As<Napi::Number>(),
+                       info[1].As<Napi::Number>().Uint32Value(),
+                       info[2].As<Napi::Number>(),
+                       info[3].As<Napi::Number>().Uint32Value());
+  return env.Undefined();
+}
+
+Napi::Value Worksheet::DataValidationCell(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+  worksheet_data_validation_cell(
+      worksheet,
+      info[0].As<Napi::Number>(),
+      info[1].As<Napi::Number>().Uint32Value(),
+      info[2].As<Napi::External<lxw_data_validation>>().Data());
+  return env.Undefined();
+}
+
+Napi::Value Worksheet::DataValidationRange(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  worksheet_data_validation_range(worksheet,
+                                  info[0].As<Napi::Number>(),
+                                  info[1].As<Napi::Number>().Uint32Value(),
+                                  info[2].As<Napi::Number>(),
+                                  info[3].As<Napi::Number>().Uint32Value(),
+                                  DataValidation(info[4]));
+
+  return env.Undefined();
+}
+
+Napi::Number callDateMethod(const Napi::Object& date, const char* method) {
+  return date.Get(method).As<Napi::Function>().Call({}).As<Napi::Number>();
+}
+
+lxw_datetime convertDate(const Napi::Object& date) {
+  return {
```

Comment:
Can't use designated initializers in C++17.

```suggestion
  lxw_datetime datetime = {};
```

Then assign the fields manually.